### ISSUE TITLE
Exception stack is bogus when ulimit is hit.

### ISSUE THREAD

**Issue by ericstj** [dotnet/coreclr#5782]
So the scenario is you have an app exceed ULIMIT on OSX.  When this is hit an IO exception is thrown.  In the process of trying to write out the stacktrace, StackFrameHelper tries to load a type in System.Diagnostics.StackTrace.dll (+1 file handle) which in turn needs to load System.Reflection.Metadata.dll (+1 file handle) in order to print the source info.  The result is that I don't get the real exception, and instead get a bogus ERROR_INTERNAL_ERROR from the CLR.

See below
```
(Exception from HRESULT: 0x8007054F)
   --- End of inner exception stack trace ---
   at System.RuntimeTypeHandle.CreateInstance(RuntimeType type, Boolean publicOnly, Boolean noCheck, Boolean& canBeCached, RuntimeMethodHandleInternal& ctor, Boolean& bNeedSecurityCheck)
   at System.RuntimeType.CreateInstanceSlow(Boolean publicOnly, Boolean skipCheckThis, Boolean fillCache, StackCrawlMark& stackMark)
  at System.Activator.CreateInstance(Type type, Boolean nonPublic)
   at System.Activator.CreateInstance(Type type)
   at System.Diagnostics.StackFrameHelper.InitializeSourceInfo(Int32 iSkip, Boolean fNeedFileInfo, Exception exception)
   at System.Diagnostics.StackTrace.CaptureStackTrace(Int32 iSkip, Boolean fNeedFileInfo, Thread targetThread, Exception e)
   at System.Environment.GetStackTrace(Exception e, Boolean needFileInfo)
   at System.Exception.GetStackTrace(Boolean needFileInfo)
   at System.Exception.ToString(Boolean needFileLineInfo, Boolean needMessage)
   at System.Exception.ToString(Boolean needFileLineInfo, Boolean needMessage)
   at System.AggregateException.ToString()
```

We should probably either a) load these up-front or b) fail gracefully if we cannot get StackTraceSymbols type.

It looks like the code was already trying to do that [here](https://github.com/dotnet/coreclr/blob/76837be7f1216bc0a970b20c30db006c9e6dacbe/src/mscorlib/src/System/Diagnostics/Stacktrace.cs#L124), unfortunately `throwOnError: false` can still throw FileLoadExceptions.

/cc @mikem8361

**Comment by mikem8361** [comment]
@tzwlai Is this something we should look at for 1.0?

**Comment by tzwlai** [comment]
What is the likelihood of an app exceeding the ULIMIT?  I believe it's something the user can configure, right?

**Comment by mikem8361** [comment]
Yes, ulimit can be set by the user. I think this doesn’t meet the Escrow bar.

**Comment by ericstj** [comment]
It's quite likely since Windows doesn't have this restriction.  The problem here isn't that the user is blocked, the problem is that the error message doesn't lead them to the right fix.

I agree that it doesn't meet the escrow bar, but I do think we should "kb" it for RTM since the error is bogus and doesn't lead towards the right solution.

**Comment by tzwlai** [comment]
Ok, we are on the same page regarding Escrow then.  I also agree that we should document this in some KB/README.  I'll follow up.

**Comment by mikem8361** [comment]
/cc: @jkotas @noahfalk 

Catching the exceptions from Type.GetType or Activator.CreateInstance doesn't really help much when running out of handles. Even if we get pass loading/creating the StackTraceSymbols class  might not be enough handles left to load/parse the pdb in System.Reflection.Metadata.dll. And if we actually get pass that there might not be enough handles to even display the stack trace string.

I tested my changes (wrapping the above two functions in try/catch PR #5802) by artificially reducing the ULIMIT down to 20 or 30 (ulimit -n 20) and in my very simple test case it got further but in the end low handles causes failures from most any code that actually does something.

Should we actually try to fix this since it will probably fail any way?

**Comment by ericstj** [comment]
You were able to dump the stack of the nested exception so you must be able to give something better than what I get today.

**Comment by mikem8361** [comment]
Fixed PR #5802.
